source/particle/property/elastic_tensor_decomposition.cc

```diff
@@ -0,0 +1,479 @@
+
+        property_information.push_back(std::make_pair("cpo elastic axis e1",3));
+        property_information.push_back(std::make_pair("cpo elastic axis e2",3));
+        property_information.push_back(std::make_pair("cpo elastic axis e3",3));
+        property_information.push_back(std::make_pair("cpo elastic hexagonal axis",3));
+        property_information.push_back(std::make_pair("cpo elastic vector norm square",1));
+        property_information.push_back(std::make_pair("cpo elastic triclinic vector norm square p1",1));
+        property_information.push_back(std::make_pair("cpo elastic triclinic vector norm square p2",1));
+        property_information.push_back(std::make_pair("cpo elastic triclinic vector norm square p3",1));
+        property_information.push_back(std::make_pair("cpo elastic monoclinic vector norm square p1",1));
+        property_information.push_back(std::make_pair("cpo elastic monoclinic vector norm square p2",1));
+        property_information.push_back(std::make_pair("cpo elastic monoclinic vector norm square p3",1));
+        property_information.push_back(std::make_pair("cpo elastic orthorhombic vector norm square p1",1));
+        property_information.push_back(std::make_pair("cpo elastic orthorhombic vector norm square p2",1));
+        property_information.push_back(std::make_pair("cpo elastic orthorhombic vector norm square p3",1));
+        property_information.push_back(std::make_pair("cpo elastic tetragonal vector norm square p1",1));
+        property_information.push_back(std::make_pair("cpo elastic tetragonal vector norm square p2",1));
+        property_information.push_back(std::make_pair("cpo elastic tetragonal vector norm square p3",1));
+        property_information.push_back(std::make_pair("cpo elastic hexagonal vector norm square p1",1));
+        property_information.push_back(std::make_pair("cpo elastic hexagonal vector norm square p2",1));
+        property_information.push_back(std::make_pair("cpo elastic hexagonal vector norm square p3",1));
+        property_information.push_back(std::make_pair("cpo elastic isotropic vector norm square",1));
+
+        return property_information;
+      }
+
+
+
+      template <int dim>
+      void
+      ElasticTensorDecomposition<dim>::declare_parameters (ParameterHandler &)
+      {}
+
+
+
+      template <int dim>
+      void
+      ElasticTensorDecomposition<dim>::parse_parameters (ParameterHandler &)
+      {}
+    }
+  }
+}
+
+
+
+// explicit instantiations
+namespace aspect
+{
+  namespace Particle
+  {
+    namespace Property
+    {
+      ASPECT_REGISTER_PARTICLE_PROPERTY(ElasticTensorDecomposition,
+                                        "elastic tensor decomposition",
+                                        "A plugin in which decomposes the elastic tensor "
+                                        "into different approximations and provides the "
+                                        "eigenvectors of the tensor.")
```

Comment:
Is it worth describing in a bit more details what the different approximations are?